plugins/node/opentelemetry-instrumentation-koa/src/types.ts

```diff
@@ -13,28 +13,9 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import type { Middleware, ParameterizedContext, DefaultState } from 'koa';
-import type { RouterParamContext } from '@koa/router';
-import type * as Router from '@koa/router';
 import { Span } from '@opentelemetry/api';
 import { InstrumentationConfig } from '@opentelemetry/instrumentation';
-
-export enum KoaLayerType {
-  ROUTER = 'router',
-  MIDDLEWARE = 'middleware',
-}
-
-export type KoaContext = ParameterizedContext<DefaultState, RouterParamContext>;
-
-export type KoaMiddleware = Middleware<DefaultState, KoaContext> & {
-  router?: Router;
-};
-
-export type KoaRequestInfo = {
-  context: KoaContext;
-  middlewareLayer: Middleware<DefaultState, KoaContext>;
-  layerType: KoaLayerType;
-};
+import { KoaLayerType, KoaRequestInfo } from './internal-types';
```

Comment:
`types.ts` cannot import from `internal-types.ts`, as this would include all private types in this file into the public api for the package. The other way is ok.

the `types.ts` should include everything public, which is the `KoaInstrumentationConfig` and all the tansitive types it references.

So `KoaLayerType`, `KoaRequestInfo` and `KoaMiddleware` are all part of the public API thus they must remain in `types.ts`. `KoaContext` is not used for anything public, thus it should be moved to `internal-types.ts`.

Reply:
`KoaRequestInfo` uses `KoaContext`, so if I understood correct, it makes `KoaContext` used in the public API, so it can't be moved to the `internal-types`.

https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-koa/src/types.ts#L34

Reply:
You are right!
I meant to write before that only `KoaMiddleware` should be moved to `internal-types.ts` and not `KoaContext`, which is what you implemented in the latest version. Thanks again 🥇